.github/workflows/CI.yml

```diff
@@ -27,6 +27,12 @@ on:
   repository_dispatch:
     types: [dependency-build, workflow-run]
 
+  workflow_dispatch:
```

Comment:
I don't think this will work since you are sending in the sha but not using it

Reply:
You are right. Adding the checkout condition to checkout the sha if it has the sha; otherwise, it is default.